File: 240420/숫자 카운트/numeric-count.py
N = int(input())
B_asks = []
for _ in range(N):
    num, cnt1, cnt2 = map(int, input().split())
    num = list(map(int, str(num)))
    B_asks.append([num[0], num[1], num[2], cnt1, cnt2])

cnts = 0
for i in range(1, 10):
    for j in range(1, 10):
        for k in range(1, 10):
            if i == j or j == k or k == i:
                continue

            b_pred = [i, j, k]
            satisfied = True
            for b_ask in B_asks:
                cnt1, cnt2 = 0, 0

                for l in range(3):
                    for n in range(3):
                        if b_pred[l] == b_ask[n]:
                            if l == n:
                                cnt1 += 1
                            else:
                                cnt2 += 1

                if cnt1 != b_ask[3] or cnt2 != b_ask[4]:
                    satisfied = False

            if satisfied:
                cnts += 1

print(cnts)

File: 260226/숫자 카운트/numeric-count.cpp
#include <bits/stdc++.h>
using namespace std;

struct Ask 
{
    int d[3];  
    int cnt1;   
    int cnt2;   
};

int main() 
{
    int N;
    cin >> N;

    vector<Ask> asks;
    asks.reserve(N);

    for (int t = 0; t < N; t++) 
    {
        int num, c1, c2;
        cin >> num >> c1 >> c2;

        string s = to_string(num); 
        Ask a;
        a.d[0] = s[0] - '0';
        a.d[1] = s[1] - '0';
        a.d[2] = s[2] - '0';
        a.cnt1 = c1;
        a.cnt2 = c2;
        asks.push_back(a);
    }

    int cnts = 0;

    for (int i = 1; i <= 9; i++) 
    {
        for (int j = 1; j <= 9; j++) 
        {
            for (int k = 1; k <= 9; k++) 
            {
                if (i == j || j == k || k == i) continue;

                int pred[3] = {i, j, k};
                bool satisfied = true;

                for (const auto& ask : asks) 
                {
                    int strike = 0, ball = 0;

                    for (int l = 0; l < 3; l++) 
                    {
                        for (int n = 0; n < 3; n++) 
                        {
                            if (pred[l] == ask.d[n]) 
                            {
                                if (l == n) strike++;
                                else ball++;
                            }
                        }
                    }

                    if (strike != ask.cnt1 || ball != ask.cnt2) 
                    {
                        satisfied = false;
                        break; 
                    }
                }

                if (satisfied) cnts++;
            }
        }
    }

    cout << cnts << "\n";
    return 0;
}